src/integrations/Sendinblue/utils.js

```diff
@@ -0,0 +1,152 @@
+import { getDefinedTraits } from "../../utils/utils";
+import {
+  NAME,
+  excludeReservedTraits,
+  excludePageProperties,
+} from "./constants";
+import {
+  getHashFromArray,
+  removeUndefinedAndNullValues,
+  isNotEmpty,
+} from "../utils/commonUtils";
+
+const validateEmail = (email) => {
+  const regex =
+    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  if (!regex.test(email)) {
+    return false;
+  }
+  return true;
+};
+
+const validatePhoneWithCountryCode = (phone) => {
+  const regex = /^\+(?:[{0-9] ?){6,14}[0-9]$/;
+  if (!regex.test(phone)) {
+    return false;
+  }
+  return true;
+};
+
+// FIRSTNAME, LASTNAME, SMS and EMAIL are the default created contact attributes in sendinblue
+const prepareDefaultContactAttributes = (message, trackCall = false) => {
+  const { firstName, lastName, phone, email } = getDefinedTraits(message);
+  if (trackCall) {
+    return {
+      FIRSTNAME: firstName,
+      LASTNAME: lastName,
+      SMS: phone,
+      EMAIL: email,
+    };
+  }
+  return { FIRSTNAME: firstName, LASTNAME: lastName, SMS: phone };
+};
+
+/**
+ * Function to remove duplicate traits from user traits
+ * @param {*} userTraits {"FIRSTNAME":"John","firstname":"John","location":"San Francisco","LOCATIOn":"San Francisco"}
+ * @param {*} attributeMap {"area_code":"AREA","location":"LOCATION"}
+ * @returns // {"FIRSTNAME":"John","LOCATIOn":"San Francisco"}
+ */
+const refineUserTraits = (userTraits, attributeMap) => {
+  const refinedTraits = userTraits;
+  Object.keys(userTraits).forEach((key) => {
+    if (
+      excludeReservedTraits.includes(key) ||
+      Object.prototype.hasOwnProperty.call(attributeMap, key)
+    ) {
+      delete refinedTraits[key];
+    }
+  });
+  return refinedTraits;
+};
+
+/**
+ * Function to prepare user traits for identify and track call.
+ * User traits is a combination of some reserved traits and transformation of reset of the traits with the help of contact attribute mapping defined in webapp.
+ * Reserved traits -> first name, last name, phone, email
+ * Contact attribute mapping -> [{"from":"area_code","to":"AREA"},{"from":"location","to":"LOCATION"}]
+ * @param {*} message
+ * @param {*} contactAttributeMapping traits to Sendinblue contact attribute mapping defined in webapp
+ * @returns
+ */
+const prepareUserTraits = (message, contactAttributeMapping, trackCall) => {
+  const { traits } = message.context;
```

Comment:
```suggestion
  const traits = message.traits || message.context.traits;
```

Reply:
In SDK, we don't have root-level traits right?
<img width="869" alt="Screenshot 2022-12-08 at 6 15 57 PM" src="https://user-images.githubusercontent.com/60897972/206450088-6c6a508b-b8f3-4132-8884-fe00acc11cf8.png">